packages/flutter_sprites/README.md

```diff
@@ -1,8 +1,209 @@
 # Flutter Sprites
+Flutter Sprites is a toolkit for building complex, high performance animations and 2D games with Flutter. Your sprite render tree lives inside a SpriteWidget that mixes seamlessly with other Flutter and Material widgets. You can use Flutter Sprites to create anything from an animated icon to a full fledged game.
 
-A sprite toolkit built on top of Flutter.
+This guide assumes a basic knowledge of Flutter and Dart. You can find an example of Flutter Sprites in the Flutter Gallery in the Weather demo, or in the flutter/game repository on Github.
 
-## Getting Started
+## Setting up a SpriteWidget
+The first thing you need to do to use Flutter Sprites is to setup a SpriteWidget with a root node that is used to draw it's contents. Any sprite nodes that you add to the root node will be rendered by the SpriteWidget. Typically, your root node is part of your app's state. This is an example of how you can setup a custom stateful widget with Flutter Sprites:
 
-For help getting started with Flutter, view our online
-[documentation](http://flutter.io).
+    import 'package:flutter/material.dart';
+    import 'package:flutter_sprites/flutter_sprites.dart';
+    
+    class MyWidget extends StatefulWidget {
+      @override
+      MyWidgetState createState() => new MyWidgetState();
+    }
+    
+    class MyWidgetState extends State<MyWidget> {
+      NodeWithSize rootNode;
```

Comment:
Please briefly describe what the NodeWithSize class is and why it's used as the type for the root node.